jack-test/test/java/com/rapleaf/jack/transaction/TestDbMetrics.java

```diff
@@ -0,0 +1,200 @@
+    Future<Long> future1 = executorService.submit(() -> {
+      long start = stopwatch.elapsedMillis();
+      transactor.execute(a -> {sleepMillis(100);});
+      return stopwatch.elapsedMillis() - start;
+    });
+    Future<Long> future2 = executorService.submit(() -> {
+      long start = stopwatch.elapsedMillis();
+      transactor.execute(a -> {sleepMillis(100);});
+      return stopwatch.elapsedMillis() - start;
+    });
+    double expectedAverageConnectionExecutionTime = ((double)future1.get() + (double)future2.get()) / 2;
+    DbMetrics dbMetrics = transactor.getDbMetrics();
+    double averageConnectionExecutionTime = dbMetrics.getAverageConnectionExecutionTime();
+    transactor.close();
+
+    assertRoughEqual(averageConnectionExecutionTime, expectedAverageConnectionExecutionTime, 20);
+  }
+
+  @Test
+  public void testAverageIdleConnections() throws Exception {
+    TransactorImpl<IDatabase1> transactor = transactorBuilder.setMaxTotalConnections(2).setMinIdleConnections(1).setKeepAliveTime(Duration.millis(50)).get();
+    transactor.execute(a -> {sleepMillis(100);});
+    long startIdleTime = stopwatch.elapsedMillis();
+    sleepMillis(100);
+    long idleTime = stopwatch.elapsedMillis() - startIdleTime;
+    DbMetrics dbMetrics = transactor.getDbMetrics();
+    long lifeTime = dbMetrics.getLifeTime();
+    double expectedAverageIdleConnections = (double)idleTime / (double)lifeTime;
+    double averageIdleConnectionsMaxValue = dbMetrics.getAverageIdleConnectionsMaxValue();
+    double averageIdleConnectionsMinValue = dbMetrics.getAverageIdleConnectionsMinValue();
+    transactor.close();
+
+    assert ((expectedAverageIdleConnections - .1 <= averageIdleConnectionsMaxValue) && (expectedAverageIdleConnections + .1 >= averageIdleConnectionsMinValue));
+  }
+
+  @Test
+  public void testAverageActiveConnections() throws Exception {
+    TransactorImpl<IDatabase1> transactor = transactorBuilder.get();
+    long startActive = stopwatch.elapsedMillis();
+    transactor.execute(db -> {sleepMillis(100);});
+    long activeTime = stopwatch.elapsedMillis() - startActive;
+    sleepMillis(100);
+    DbMetrics dbMetrics = transactor.getDbMetrics();
+    long lifeTime = dbMetrics.getLifeTime();
+    double expectedAverageActiveConnections = (double)activeTime / (double)lifeTime;
+    double averageActiveConnections = dbMetrics.getAverageActiveConnections();
+    transactor.close();
+    assertRoughEqual(expectedAverageActiveConnections, averageActiveConnections, .2);
+  }
+
+  @Test
+  public void testLifeTime() throws Exception {
+    long startTime = stopwatch.elapsedMillis();
+    TransactorImpl<IDatabase1> transactor = transactorBuilder.get();
+    sleepMillis(100);
+    DbMetrics dbMetrics = transactor.getDbMetrics();
+    assertRoughEqual(stopwatch.elapsedMillis() - startTime, dbMetrics.getLifeTime(), 20);
+  }
+
+  private void assertRoughEqual(double value, double expected, double error) {
```

Comment:
Never mind. Some of the values passed into this method are `long`s. So using `assertEquals` will need type conversion, which is not very convenient.